main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/GLibLogging.cs

```diff
@@ -268,13 +270,32 @@ static void LoggerMethod (IntPtr logDomainPtr, LogLevelFlags logLevel, IntPtr me
 			case LogLevelFlags.Error:
 			case LogLevelFlags.Critical:
 			default:
-				LoggingService.LogError (msg);
+				var exception = new CriticalGtkException (message, stackTraceString);
+				if (logLevel.HasFlag (LogLevelFlags.FlagFatal))
+					LoggingService.LogFatalError ("Fatal GLib error", exception);
```

Comment:
Why pass a static string to either of these methods?  I would have expected:

```suggestion
					LoggingService.LogFatalError (msg, exception);
				else
					LoggingService.LogInternalError (msg, exception);
```

This is more in line with the one for obj-c in the previous file.  I worry that setting the string here will end up putting the wrong message in telemetry, but I'm not 100% sure how all of that plumbs through.

Reply:
The static message is for our log files only:
http://source.monodevelop.com/#MonoDevelop.Core/MonoDevelop.Core/LoggingService.cs,572